channeldb/payment_control.go

```diff
@@ -188,6 +200,14 @@ func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
 		var check []inflightCheck
 		check = append(check, ensureNonTerminal())
 
+		// We must also check that registering this attempt doesn't
+		// take the total sent amount above the payment value.
+		if info.Value < attempt.Route.ReceiverAmt() {
+			return ErrValueExceedsAmt
+		}
```

Comment:
If we do validation here, we should probably also check that the mpp payment addresses and total amounts match with the in-flight routes? Or is that too strict for sendtoroute?

Reply:
Added validation + tests!